src/storage/recovery/recovery_manager.cpp

```diff
@@ -0,0 +1,762 @@
+
+            // Step 5: Rename the table
+            auto result UNUSED_ATTRIBUTE = GetDatabaseCatalog(txn, next_redo_record->GetDatabaseOid())
+                                               ->RenameTable(txn, catalog::table_oid_t(next_class_oid), name_string);
+            TERRIER_ASSERT(result, "Renaming should always succeed during replaying");
+
+            // Step 6: Update metadata and clean up additional record processed. We need to use the indexes on
+            // pg_class to find what tuple slot we just inserted into. We get the new tuple slot using the oid index.
+            auto pg_class_oid_index = GetDatabaseCatalog(txn, next_redo_record->GetDatabaseOid())->classes_oid_index_;
+            auto pr_init = pg_class_oid_index->GetProjectedRowInitializer();
+            buffer = common::AllocationUtil::AllocateAligned(pr_init.ProjectedRowSize());
+            pr = pr_init.InitializeRow(buffer);
+            *(reinterpret_cast<uint32_t *>(pr->AccessForceNotNull(0))) = next_class_oid;
+            std::vector<TupleSlot> tuple_slot_result;
+            pg_class_oid_index->ScanKey(*txn, *pr, &tuple_slot_result);
+            TERRIER_ASSERT(tuple_slot_result.size() == 1, "Index scan should only yield one result");
+            tuple_slot_map_[next_redo_record->GetTupleSlot()] = tuple_slot_result[0];
+            delete[] buffer;
+            tuple_slot_map_.erase(delete_record->GetTupleSlot());
+            delete[] reinterpret_cast<byte *>(next_redo_record);
+
+            return 1;  // We processed an additional record
+          }
+        }
+      }
+    }
+
+    // Step 3: If it was not a renaming, we call to the catalog to delete the class object
+    bool result UNUSED_ATTRIBUTE;
+    if (class_kind == catalog::postgres::ClassKind::REGULAR_TABLE) {
+      result =
+          GetDatabaseCatalog(txn, delete_record->GetDatabaseOid())->DeleteTable(txn, catalog::table_oid_t(class_oid));
+    } else if (class_kind == catalog::postgres::ClassKind::INDEX) {
+      result =
+          GetDatabaseCatalog(txn, delete_record->GetDatabaseOid())->DeleteIndex(txn, catalog::index_oid_t(class_oid));
+    } else {
+      TERRIER_ASSERT(false, "We only support replaying of dropping of tables and indexes");
+    }
+    TERRIER_ASSERT(result, "Table/index DROP should always succeed");
+
+    // Step 5: Clean up metadata
+    tuple_slot_map_.erase(delete_record->GetTupleSlot());
+
+    return 0;  // No additional logs processed
+  }
+
+  // A delete into pg_attribute means we are deleting a column. There are two cases:
+  //  1. Drop column: This requires some additional processing to actually clean up the column
+  //  2. Cascading delete from drop table: In this case, we don't process the record because the DeleteTable catalog
+  //  function will clean up the columns
+  if (delete_record->GetTableOid() == catalog::COLUMN_TABLE_OID) {
+    return 0;  // Case 2, no additional records processed
+  }
```

Comment:
What this is doing is kinda implicit, I should add a comment here. By doing nothing in this conditional, we implicitly skip this record, so we don't actually delete the entry in `pg_attribute`. 

The comment above the code explains why we do that, which is to let the catalog delete the entries. Otherwise, an assert exception gets thrown in the catalog saying something like "You're trying to delete a table/index with no columns"

Reply:
That's weird.  I just scanned through the code looking for an assert along the lines of what you say you saw.  Do you have a line number for the triggered assert?

Reply:
That may actually be a logic bug in `DeleteColumns`.  I'm not sure why we decided to return false when its empty (generally used to indicate a must-abort situation), and the Doxygen comment on the return value is definitely wrong:  Apparently the boolean is a column from pg_attribute...

I'll submit a PR to fix the issues to master here in a second, and that should allow you to avoid handling cascaded deletes with a special case.